### PR TITLE
C# SDK change with Hub's certificate and operation tasks

### PR DESCRIPTION
**Pull request by anusapan**
## Description
Regenerated the SDK based off a new swagger and add/update the test cases for updated SDK.
[Updated swagger PR](https://github.com/Azure/azure-rest-api-specs/commit/9bc154a142ae39c3b900ba91043aabed1ef8e2f3)

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: NuGet.Config [hunk 1]
```diff
@@ -1,8 +1,7 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <configuration>
   <packageSources>
-     <!-- To enable LocalFeed for testing uncomment the following line -->
-     <add key="NugetOfficialV3" value="https://api.nuget.org/v3/index.json" />
-     <add key="Local" value="tools/LocalNugetFeed" />
-   </packageSources>
+    <add key="NugetOfficialV3" value="https://api.nuget.org/v3/index.json" />
```
Comment (andrew-buckley) [inline NuGet.Config:4]: Why this change?

File: src/SDKs/IotHub/IotHub.Tests/Helpers/IotHubTestUtilities.cs [hunk 2]
```diff
@@ -21,6 +21,22 @@ public class IotHubTestUtilities
         public static string DefaultResourceGroupName = "DotNetHubRG";
         public static string DefaultUpdateResourceGroupName = "UpdateDotNetHubRG";
         public static string EventsEndpointName = "events";
+        public static string DefaultCertificateIotHubName = "DotNetCertificateHub";
```
Comment (andrew-buckley) [inline src/SDKs/IotHub/IotHub.Tests/Helpers/IotHubTestUtilities.cs]: Might be better to programmatically generate this certificate.

**Review by andrew-buckley — APPROVED** [review]
You might want to also squash your commits before making the next PR.

File: src/SDKs/IotHub/IotHub.Tests/Helpers/IotHubTestUtilities.cs (hunk 2 above)
Comment (anusapan) [reply src/SDKs/IotHub/IotHub.Tests/Helpers/IotHubTestUtilities.cs]: OpenSsl is not compatible with netcoreapp1.1
